scheduler/feasible.go

```diff
@@ -223,11 +235,17 @@ func (h *HostVolumeChecker) hasVolumes(n *structs.Node) bool {
 			}
 
 			if req.Sticky {
-				if slices.Contains(h.hostVolumeIDs, vol.ID) || len(h.hostVolumeIDs) == 0 {
+				// the node is feasible if there are no claims or if there's an
+				// exact match
+				if len(h.claims) == 0 {
 					return true
 				}
 
-				return false
+				for _, c := range h.claims {
+					if c.VolumeID == vol.ID {
+						return true
+					}
+				}
```

Comment:
We still need to handle the case where the job is increased in size. Consider this case:

* we have a matching volume on nodes A and B.
* job v1 has alloc 1 which claims the volume on node A
* job v2 makes a destructive update and changes `group.count = 2`
* alloc1' considers node A, finds the volume claim on node A so it's placed there
* alloc2' considers node B, finds we have claims > 0 so it skips node B
* alloc2' find the _same_ volume claim on node A so it's placed there

Each time we pick a node here, we need to remove or otherwise mark the claims we've already seen, so that the next alloc considered in this evaluation can get an unclaimed volume.

I'm pretty sure you can handle this by removing the claim from the list.

Reply:
~Yeah... no... I think it's a bit more tricky than this. If we remove claims from the table, allocs that get migrated won't know they had a claimed volume before and we're back to square one.~ Ah wait, you mean remove from the list in the `HostVolumeChecker` object? Oh right that would maybe make more sense because we populate that list in the stack creation, but `Feasible` gets cold on every iteration. Yeah ok I think I get it.

Reply:
[aa56c30](https://github.com/hashicorp/nomad/pull/24993/commits/aa56c30788f66995b5405bfef36d8d61b8dcfd40)

Reply:
a *clever* test will follow